LambdaEngine/Include/Rendering/PaintMaskRenderer.h

```diff
@@ -85,32 +99,35 @@ namespace LambdaEngine
 		TSharedRef<CommandAllocator>	m_CopyCommandAllocator = nullptr;
 		TSharedRef<CommandList>			m_CopyCommandList = nullptr;
 
-		CommandAllocator** m_ppRenderCommandAllocators = nullptr;
-		CommandList** m_ppRenderCommandLists = nullptr;
+		CommandAllocator**			m_ppRenderCommandAllocators = nullptr;
+		CommandList**				m_ppRenderCommandLists = nullptr;
 
 		uint64						m_PipelineStateID = 0;
 		TSharedRef<PipelineLayout>	m_PipelineLayout = nullptr;
 		TSharedRef<DescriptorHeap>	m_DescriptorHeap = nullptr;
 
-		GUID_Lambda m_VertexShaderGUID = 0;
-		GUID_Lambda m_PixelShaderGUID = 0;
+		GUID_Lambda					m_VertexShaderGUID = 0;
+		GUID_Lambda					m_PixelShaderGUID = 0;
 
-		TSharedRef<RenderPass> m_RenderPass = nullptr;
+		TSharedRef<RenderPass>		m_RenderPass = nullptr;
 
-		TSharedRef<Sampler> m_Sampler = nullptr;
-		TArray<TSharedRef<Buffer>> m_TransformCopyBuffers;
-		TSharedRef<Buffer> m_TransformBuffer = nullptr;
+		TSharedRef<Sampler>			m_Sampler = nullptr;
+		TArray<TSharedRef<Buffer>>	m_UnwrapDataCopyBuffers;
+		TSharedRef<Buffer>			m_UnwrapDataBuffer = nullptr;
 
 		const DrawArg*												m_pDrawArgs;
 		TArray<TArray<TSharedRef<DescriptorSet>>>					m_VerticesInstanceDescriptorSets;
-		//TArray<TArray<TArray<TSharedRef<DescriptorSet>>>>			m_TransformDescriptorSets;
 
-		std::optional<TSharedRef<DescriptorSet>>					m_PerFrameBufferDescriptorSets;
+		std::optional<TSharedRef<DescriptorSet>>					m_UnwrapDataDescriptorSet;
+		std::optional<TSharedRef<DescriptorSet>>					m_PerFrameBufferDescriptorSet;
 		std::optional<TSharedRef<DescriptorSet>>					m_BrushMaskDescriptorSet;
```

Comment:
Isn't an optional pointer unnecessary? You could just check for `nullptr`

Reply:
It is completely unnecessary! fixed